extensions/MESA/EGL_MESA_config_select_group.txt

```diff
@@ -0,0 +1,108 @@
+Name
+
+    EGL_MESA_config_select_group
+
+Name Strings
+
+    EGL_MESA_config_select_group
+
+Contributors
+
+    Hal Gentz <zegentzy@protonmail.com>
+    Adam Jackson <ajax@redhat.com>
+    Robert Mader <robert.mader@posteo.de>
+
+Contacts
+
+    Hal Gentz <zegentzy@protonmail.com>
+
+Status
+
+    Draft
+
+Version
+
+    Version 3, 2020-04-06
+
+Number
+
+    EGL Extension #143
+
+Extension Type
+
+    EGL client extension
+
+Dependencies
+
+    This extension is written against the wording of the 2014.08.27 revision
+    of the EGL 1.5 Specification.
+
+Overview
+
+    This extension provides a mechanism to lower the config selection priority
+    of the configs returned by eglChooseConfig. A new config attribute is
+    introduced, providing a way for the implementation to group configs into
+    different config selection categories.  The config selection priorities of
+    all the configs in one category may be higher or lower than the ones in
+    other categories.  The config selection priorities in one category follow
+    the rules of eglChooseConfig.
+
+New Types
+
+    None
+
+New Procedures and Functions
+
+    None
+
+New Tokens
+
+    Accepted as a new EGLConfig attribute:
+
+        EGL_CONFIG_SELECT_GROUP_MESA                0x31DF
+
+Additions to the EGL 1.5 Specification
+
+    In section 3.4 "Configuration Management", under the "Buffer Descriptions
+    and Attributes" heading, append to table 3.1 the following:
+
+    "EGL_CONFIG_SELECT_GROUP_MESA | integer | the config select group"
+
+    In section 3.4 "Configuration Management", under the "Other EGLConfig
+    Attribute Descriptions" heading, append to the end the following:
+
+    "The attribute EGL_CONFIG_SELECT_GROUP_MESA is used to specify the config
+    selection category of the config. Configs in the i-th category have a higher
+    config selection priority than those found in the (i+1)-th category.
+
+    "The attribute EGL_CONFIG_SELECT_GROUP_MESA is used by the implementation to
+    override the default sorting rules, by collecting configurations into
+    groups. It is not accepted by the <attrib_list> parameter of
```

Comment:
Not accepted with eglChooseConfig, but you are providing details on selection?

Reply:
This was at my suggestion, primarily for diagnosing why you got the config you did. I'm not sure it's the best plan, the corresponding GLX extn doesn't let you get this from the app, it's just a protocol detail. I'd be just as happy to drop that bit I think.

Reply:
The justification seems fine IMHO. I can't think of use cases outside of automated testing and printf debugging, but those are real use cases.